src/Pate/Verification/Widening.hs

```diff
@@ -171,31 +179,123 @@ widenAlongEdge bundle from d gr to = withSym $ \sym ->
+        Just c <- lift $ withOnlineBackend $ \bak -> withSolverProcess $ \sp ->
+          liftIO $ W4.inNewFrame sp $ do
+            W4.assume (W4.solverConn sp) asm
+            W4.asConcrete <$>
+              PVC.resolveSingletonSymbolicAsDefault bak (PS.unSE se)
+        liftIO $ PS.concreteScope @v2 sym c
+
+      asStackOffset :: forall bin tp. PBi.WhichBinaryRepr bin -> PS.ScopedExpr sym tp pre -> MaybeT (EquivM_ sym arch) (PS.ScopedExpr sym tp post)
+      asStackOffset bin se = do
+        W4.BaseBVRepr w <- return $ W4.exprType (PS.unSE se)
+        Just Refl <- return $ testEquality w (MM.memWidthNatRepr @(MM.ArchAddrWidth arch))
+
+        -- se[v]
+        let postFrame = PS.simStackBase $ PS.simVarState $ (PPa.getPair' bin postVars)
+        off <- liftIO $ PS.liftScope0 @post sym (\sym' -> W4.freshConstant sym' (W4.safeSymbol "frame_offset") (W4.BaseBVRepr w))
+        -- asFrameOffset := frame[post] + off
+        asFrameOffset <- liftIO $ PS.liftScope2 sym W4.bvAdd postFrame off
+        -- asFrameOffset' := frame[post/f(v)] + off
+        asFrameOffset' <- liftIO $ PS.applyScopeCoercion sym post_to_pre asFrameOffset
+        -- asm := se == frame[post/f(pre)] + off
+        asm <- liftIO $ PS.liftScope2 sym W4.isEq se asFrameOffset'
+        -- assuming 'asm', is 'off' constant?
+        off' <- asScopedConst (PS.unSE asm) off
+        lift $ traceBundle bundle (show $ W4.printSymExpr (PS.unSE off'))
+        -- return frame[post] + off
+        liftIO $ PS.liftScope2 sym W4.bvAdd postFrame off'
+
+      asSimpleAssign :: forall tp. PS.ScopedExpr sym tp pre -> MaybeT (EquivM_ sym arch) (PS.ScopedExpr sym tp post)
+      asSimpleAssign se = do
+        -- se[pre]
+        -- se' := se[pre/post]
+        se' <- liftIO $ PS.applyScopeCoercion sym pre_to_post se
+        -- e'' := se[post/f(pre)]
+        e'' <- liftIO $ PS.applyScopeCoercion sym post_to_pre se'
+        -- se is the original value, and e'' is the value rewritten
+        -- to be phrased over the post-state
+        heuristicTimeout <- lift $ CMR.asks (PC.cfgHeuristicTimeout . envConfig)
+        asm <- liftIO $ PS.liftScope2 sym W4.isEq se e''
+        True <- lift $ isPredTrue' heuristicTimeout (PS.unSE asm)
+        return se'
+
+    PS.scopedExprMap sym postResult $ \(se :: PS.ScopedExpr sym tp pre) -> do
+      mres <- runMaybeT $
+            asScopedConst (W4.truePred sym) se
+        <|> asStackOffset PBi.OriginalRepr se
+        <|> asStackOffset PBi.PatchedRepr se
+        <|> asSimpleAssign se
+      case mres of
+        Just se' -> return se'
+        -- FIXME: add graceful failure mode
+        Nothing -> do
+          se' <- liftIO $ PS.applyScopeCoercion sym pre_to_post se
+          e'' <- liftIO $ PS.applyScopeCoercion sym post_to_pre se'
+          curAsms <- currentAsm
+
+          fail $ "Unable to rescope:" ++ (show curAsms) ++ (show (W4.printSymExpr (PS.unSE se))) ++ "***" ++ (show (W4.printSymExpr (PS.unSE e'')))
```

Comment:
Is it worth just making an exception for this now?

Reply:
I have an incoming patch that handles this more carefully by emitting warning and dropping the element from the domain, so this 'fail' will go away once that is in place.